Sources/protoc-gen-swift/MessageFieldGenerator.swift

```diff
@@ -473,7 +466,7 @@ struct MessageFieldGenerator {
     }
 
     func generateHasProperty(printer p: inout CodePrinter, usesHeapStorage: Bool) {
-        if isRepeated || isMap || oneof != nil || (isProto3 && !isMessage) {
+        if !hasFieldPresence {
```

Comment:
Nit: `guard hasFieldPresence else { return }`? (I like to use guard for early-returns when possible.)